### PR TITLE
feat: desktop support

### PR DESCRIPTION
**Pull request by pr-Mais**
Linux & Windows support.

## Packages
- [x] firebase_core
- [x] firebase_auth
- [x] firebase_functions (cloud_functions)
- [ ] firebase_remote_config
- [ ] firebase_firestore (cloud_firestore)
- [ ] firebase_performance
- [ ] firebase_analytics
- [ ] firebase_app_check
- [ ] firebase_storage
- [ ] firebase_in_app_messaging

### CODE REVIEW

File: packages/firebase_auth/firebase_auth/pubspec.yaml [hunk 1]
```diff
@@ -36,6 +37,12 @@ flutter:
...
         pluginClass: FLTFirebaseAuthPlugin
       macos:
-        pluginClass: FLTFirebaseAuthPlugin
+        # pluginClass: FLTFirebaseFunctionsPlugin
+        # Temporary for development of Desktop packages
+        default_package: firebase_core_desktop
+      linux:
+        default_package: firebase_core_desktop
+      windows:
+        default_package: firebase_core_desktop
```
Comment (Salakar) [inline packages/firebase_auth/firebase_auth/pubspec.yaml:46]: I think these are incorrect maybe, should they be this instead:

```suggestion
        default_package: firebase_auth_desktop
      linux:
        default_package: firebase_auth_desktop
      windows:
        default_package: firebase_auth_desktop
```

File: packages/cloud_functions/cloud_functions/pubspec.yaml [hunk 2]
```diff
@@ -33,6 +33,12 @@ flutter:
...
         pluginClass: FLTFirebaseFunctionsPlugin
       macos:
-        pluginClass: FLTFirebaseFunctionsPlugin
+        # pluginClass: FLTFirebaseFunctionsPlugin
+        # Temporary for development of Desktop packages
+        default_package: firebase_core_desktop
+      linux:
+        default_package: firebase_core_desktop
+      windows:
+        default_package: firebase_core_desktop
```
Comment (Salakar) [inline packages/cloud_functions/cloud_functions/pubspec.yaml:42]:
```suggestion
        default_package: flutterfire_functions_desktop
      linux:
        default_package: flutterfire_functions_desktop
      windows:
        default_package: flutterfire_functions_desktop
```

Comment (pr-Mais) [reply packages/cloud_functions/cloud_functions/pubspec.yaml:42]: `firebase_core_desktop` is correct, the dart package starts with `flutterfire `

File: packages/firebase_auth/firebase_auth/pubspec.yaml (hunk 1 above)
Comment (pr-Mais) [reply packages/firebase_auth/firebase_auth/pubspec.yaml:46]: actually they have to be `firebase_auth_desktop`

File: packages/cloud_functions/cloud_functions/pubspec.yaml (hunk 2 above)
Comment (Salakar) [reply packages/cloud_functions/cloud_functions/pubspec.yaml:42]: Shouldn't this be `firebase_functions_desktop` instead then - this is the functions package not core 😅

Comment (pr-Mais) [reply packages/cloud_functions/cloud_functions/pubspec.yaml:42]: ops, correct

**Pull request closed without merging**